contracts/common/CommonStorage.sol

```diff
@@ -18,18 +18,29 @@
 pragma solidity 0.7.3;
 
 import "./../../lib/dappsys/auth.sol";
+import "./../common/MetaTransactionMsgSender.sol";
 
 // ignore-file-swc-131
 // ignore-file-swc-108
 
 
-contract CommonStorage is DSAuth {
+abstract contract CommonStorage is DSAuth, MetaTransactionMsgSender {
   uint256 constant UINT256_MAX = 2**256 - 1;
 
   uint256 constant AUTHORITY_SLOT = 0;
   uint256 constant OWNER_SLOT = 1;
   uint256 constant RESOLVER_SLOT = 2;
 
+  bytes32 constant PROTECTED = keccak256("Recovery Mode Protected Slot");
+
+  function protectSlot(uint256 _slot) internal always {
+    uint256 flagSlot = uint256(keccak256(abi.encodePacked("RECOVERY_PROTECTED", _slot)));
```

Comment:
Why not just hash the slot itself?

Reply:
That would collide with actual storage in the event that we wanted to protect the length of an array.